isis/src/control/objs/BundleUtilities/BundleObservationSolveSettings.cpp

```diff
@@ -47,38 +47,157 @@ namespace Isis {
+    // ACCELERATION, or a higher order polynomial
+    QString ssolve = "NONE";
+    ssolve = (QString)scParameterGroup.findKeyword("SPSOLVE");
+    ssolve = ssolve.toUpper();
+    if (ssolve == "NONE") {
+      m_instrumentPositionSolveOption = NoPositionFactors;
+      m_numberCamPosCoefSolved = 0;
+    }
+    else if (ssolve == "POSITION") {
+      m_instrumentPositionSolveOption = PositionOnly;
+      m_numberCamPosCoefSolved = 1;
+    }
+    else if (ssolve == "VELOCITIES") {
+      m_instrumentPositionSolveOption = PositionVelocity;
+      m_numberCamPosCoefSolved = 2;
+    }
+    else if (ssolve == "ACCELERATIONS") {
+      m_instrumentPositionSolveOption = PositionVelocityAcceleration;
+      m_numberCamPosCoefSolved = 3;
+    }
+    else if (csolve == "ALL"){
+      m_instrumentPositionSolveOption = AllPositionCoefficients;
+      m_numberCamPosCoefSolved = m_spkSolveDegree + 1;
+    }
+
+    m_positionAprioriSigma.clear();
+
+    // If TWIST is not specified, then a default of YES is used
+    if (scParameterGroup.hasKeyword("TWIST")) {
+      QString parval = (QString)scParameterGroup.findKeyword("TWIST");
+      parval = parval.toUpper();
+      if (parval == "TRUE" || parval == "YES") { // is this necessary ??? i think pvl and toString can handle it...
+        m_solveTwist = true;
+      }
+      else if (parval == "FALSE" || parval == "NO") {
+        m_solveTwist = false;
+      }
+      else {
+        QString msg = "The TWIST parameter must be set to TRUE or FALSE; YES or NO";
+        throw IException(IException::User, msg, _FILEINFO_);
+      }
+    }
+    // If OVERHERMITE is not specified, then a default of NO is used
+    if (scParameterGroup.hasKeyword("OVERHERMITE")) {
+     QString parval = (QString)scParameterGroup.findKeyword("OVERHERMITE");
+     parval = parval.toUpper();
+     if (parval == "TRUE" || parval == "YES") {
+       m_solvePositionOverHermiteSpline = true;
+       m_positionInterpolationType = SpicePosition::PolyFunctionOverHermiteConstant;
+     }
+     else if (parval == "FALSE" || parval == "NO") {
+       m_solvePositionOverHermiteSpline = false;
+       m_positionInterpolationType = SpicePosition::PolyFunction;
+     }
+     else {
+       QString msg = "The OVERHERMITE parameter must be set to TRUE or FALSE; YES or NO";
+       throw IException(IException::User, msg, _FILEINFO_);
+     }
+    }
+
+    if (csolve != "NONE") {
+     if (scParameterGroup.hasKeyword("CAMERA_ANGLES_SIGMA"))
+       m_anglesAprioriSigma.append((double)(scParameterGroup.findKeyword("CAMERA_ANGLES_SIGMA")));
+     if (scParameterGroup.hasKeyword("CAMERA_ANGULAR_VELOCITY_SIGMA"))
+       m_anglesAprioriSigma.append(
+           (double)(scParameterGroup.findKeyword("CAMERA_ANGULAR_VELOCITY_SIGMA")));
+     if (scParameterGroup.hasKeyword("CAMERA_ANGULAR_ACCELERATION_SIGMA"))
+       m_anglesAprioriSigma.append(
+           (double)(scParameterGroup.findKeyword("CAMERA_ANGULAR_ACCELERATION_SIGMA")));
+    }
```

Comment:
This should also look for `ADDITIONAL_CAMERA_POINTING_SIGMAS` and append whatever is passed in to `m_anglesAprioriSigma`